### PR TITLE
Refactor contracts and compile preinstalled contracts from source

### PR DESCRIPTION
**Pull request by edfelten**
* Refactor / rename the contracts and interfaces used by ArbOS, to rationalize naming and directory structure.
* Compile preinstalled contracts (ArbInfo and the ERC20/ERC721 templates) from Solidity source, rather than hardcoding their byte code into the Rust source code.
* Update ERC20/721 template contracts for compatibility with the latest version of OpenZeppelin.

### CODE REVIEW

**Review by hkalodner — CHANGES_REQUESTED** [review]
The open zeppelin repo should be checked out to a particular release tag rather than the current head. Other than that and the one small comment, this looks good

File: src/evm/abi.rs [hunk 1]
```diff
@@ -394,7 +394,7 @@ impl AbiForContract {
 
 #[test]
 fn test_function_short_signature_correct() {
-    let abi = AbiForContract::new_from_file("contracts/add/build/contracts/ArbSys.json").unwrap();
+    let abi = AbiForContract::new_from_file("contracts/arbos/build/contracts/ArbSys.json").unwrap();
```
[the remark this answers is not in the document]
Comment (edfelten) [reply src/evm/abi.rs]: Fixed

**Comment by edfelten** [comment]
> The open zeppelin repo should be checked out to a particular release tag rather than the current head. Other than that and the one small comment, this looks good

Pinned it to version 3.3.0.

**Review by hkalodner — DISMISSED** [review]
LGTM

**Comment by edfelten** [comment]
Merging this as Harry and I are both OK with the latest changes.

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**